connectors/sources/sharepoint_online.py

```diff
@@ -511,6 +536,37 @@ def _validate_sharepoint_rest_url(self, url):
     async def close(self):
         await self._http_session.close()
 
+    async def site_drive_items_delta(self, url):
+        async for response in self._graph_api_client.scroll_with_delta_link(url):
+            delta_link = (
+                response[DELTA_LINK_KEY] if DELTA_LINK_KEY in response else None
+            )
+            if "value" in response and len(response["value"]) > 0:
+                yield DriveItemsPage(response["value"], delta_link)
```

Comment:
I'd move this method to be right above `drive_items` and rename to `drive_items_delta`